### PR TITLE
#12265 fixed unclosed stream

### PR DESCRIPTION
**Pull request by jtesser**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publisher/pusher/PushUtils.java [hunk 1]
```diff
@@ -76,8 +76,10 @@ private static void addFilesToCompression(TarArchiveOutputStream taos, File file
 				if (file.isFile()) {
 			        // Add the file to the archive
 					BufferedInputStream bis = new BufferedInputStream(new FileInputStream(file));
-					IOUtils.copy(new FileInputStream(file), taos);
+					FileInputStream in = new FileInputStream(file);
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/publisher/pusher/PushUtils.java]: is it the same FileInputStream used in line 78? If so, the object can be reused. 
I suggest to handle input stream this way: https://dzone.com/articles/fileinputstream-fileoutputstream-considered-harmful

Comment (brentgriffin) [inline dotCMS/src/main/java/com/dotcms/publisher/pusher/PushUtils.java]: yeah, don't think we need both "bis" and "in" - right?

Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/pusher/PushUtils.java]: Only "in" is used, and by using "in" and not "bis" we are losing the buffer that BufferedInputStream does, however, IOUtils.copy does buffering internally (4096 vs 8192 of BufferedInputStream)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publisher/pusher/PushUtils.java [hunk 2]
```diff
@@ -75,10 +75,10 @@ private static void addFilesToCompression(TarArchiveOutputStream taos, File file
 	    			taos.putArchiveEntry(new TarArchiveEntry(file, dir + "/" + file.getName()));
 				if (file.isFile()) {
 			        // Add the file to the archive
-					BufferedInputStream bis = new BufferedInputStream(new FileInputStream(file));
-					IOUtils.copy(new FileInputStream(file), taos);
-					taos.closeArchiveEntry();
-					bis.close();
+					try(BufferedInputStream bis = new BufferedInputStream(new FileInputStream(file))) {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/publisher/pusher/PushUtils.java]: Why would we not do a 
`Files.newInputStream(file.toPath()` knowing the Memory and GC issues with FileInputStreams (even when they are closed properly)

**Pull request merged**